src/providers/CartInfoProvider.js

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useState } from 'react'
+
+export const CartInfoContext = createContext()
+
+
+// all info relating to the cart
+export const CartInfoProvider = (props) => {
+
+    const confirmedCartOrders = JSON.parse(localStorage.getItem("confirmed orders")) === null ? [{
+        id: null,
+        restaurantName: null,
+        infoOfMainMeatItem: null,
+        confirmedAddOnsToOrder: null,
+        allAddOns: null,
+        confirmedOrderQuantity: 0,
+        totalMeatPrice: 0,
+        totalConfirmedAddOnPrice: 0,
+        totalOrderPrice: 0
+    }] : JSON.parse(localStorage.getItem("confirmed orders"));
+
+    const cartPriceTotal = confirmedCartOrders.map((order) => parseFloat(order.totalOrderPrice)).reduce((price1, priceN) => (price1 + priceN));
+
+    const cartQuantityTotal = confirmedCartOrders.map((order) => order.confirmedOrderQuantity).reduce((numN, numNMinus1) => numN + numNMinus1);
+
+    const [confirmedOrdersInfo, setConfirmedOrdersInfo] = useState(confirmedCartOrders);
```

Comment:
There is no point in having 2 states when one is a derivative of another.
Make the calculation once the original state changes.
What is `numNMinus1`? A very unclear name.